### PR TITLE
Windows: detect stack overflows on non-main `Fiber`s

### LINKED ISSUE #13192: Windows cannot detect stack overflows on spawned fibers
Example: (taken from `spec/std/kernel_spec.cr:279`)

```crystal
def foo
  y = StaticArray(Int8, 512).new(0)
  foo
end

spawn do
  foo
end

sleep 60.seconds
```

This should show a stack overflow error and print a stack trace, just as when `foo` is called on the main fiber, but doesn't.

The error code here is `EXCEPTION_ACCESS_VIOLATION`, instead of `EXCEPTION_STACK_OVERFLOW`, so it looks like the OS's overflow detection does not work on those `VirtualAlloc`'ed fiber stacks.

### PR DESCRIPTION
**Pull request by HertzDevil**
Resolves #13192.

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]
The system info should not change during the runtime of a process, and consequently neither the value of `total_reserved`. So I'm wondering if it makes sense to calculate that only once and put it in a constant?

File: src/crystal/system/win32/fiber.cr [hunk 1]
```diff
@@ -1,19 +1,47 @@
...
+    # * guard (a single page), detects stack overflows
+    # * the actual usable stack (*stack_size*), grows upwards
+    #
+    # Only the usable region is returned, and this stack is treated like the
+    # ones on main fibers; a non-main `Fiber` never needs to know about the
+    # existence of the reserved area. Stack overflow detection works because
+    # Windows will fail to allocate a new guard page for these fiber stacks.
 
-    if memory_pointer.null?
+    LibC.GetNativeSystemInfo(out system_info)
```
Comment (Sija) [inline src/crystal/system/win32/fiber.cr:27]: This looks unused.

File: src/crystal/system/win32/fiber.cr [hunk 2]
```diff
@@ -1,19 +1,47 @@
...
+require "c/sysinfoapi"
 require "c/winnt"
 
 module Crystal::System::Fiber
+  RESERVED_STACK_SIZE = LibC::DWORD.new(0x10000)
+
+  private class_getter page_size : LibC::DWORD do
+    LibC.GetNativeSystemInfo(out system_info)
+    system_info.dwPageSize
+  end
```
Comment (straight-shoota) [inline src/crystal/system/win32/fiber.cr:11]: Is there a reason for caching only `page_size`, not `page_size + RESERVED_STACK_SIZE`? The difference is probably negligible, just wondering...

Comment (straight-shoota) [reply src/crystal/system/win32/fiber.cr:11]: Also, I'm not sure if lazy evaluation (i.e. `getter` with block) is really the best choice here. We should expect this value to be used in practically every Crystal program. So fetching/calculating it eagerly might reduce some performance overhead.

**Pull request merged**